### ISSUE TITLE
Exception when running kitchen converge

### ISSUE THREAD

**Issue by thedrow** [saltstack/kitchen-salt#4]
```
I, [2014-05-21T19:31:06.798440 #16015]  INFO -- Kitchen: -----> Starting Kitchen (v1.2.1)
I, [2014-05-21T19:31:07.036264 #16015]  INFO -- Kitchen: -----> Creating <default-ubuntu-1204>...
I, [2014-05-21T19:32:41.329813 #16576]  INFO -- Kitchen: -----> Starting Kitchen (v1.2.1)
I, [2014-05-21T19:32:41.578659 #16576]  INFO -- Kitchen: -----> Creating <default-ubuntu-1404>...
I, [2014-05-21T19:37:36.812231 #4255]  INFO -- Kitchen: -----> Starting Kitchen (v1.2.1)
I, [2014-05-21T19:37:37.050051 #4255]  INFO -- Kitchen: -----> Creating <default-ubuntu-server-1404>...
E, [2014-05-21T19:37:40.861568 #4255] ERROR -- Kitchen: ------Exception-------
E, [2014-05-21T19:37:40.861688 #4255] ERROR -- Kitchen: Class: Kitchen::ActionFailed
E, [2014-05-21T19:37:40.861747 #4255] ERROR -- Kitchen: Message: Failed to complete #create action: [Expected process to exit with [0], but received '1'
---- Begin output of vagrant up --no-provision --provider=virtualbox ----
STDOUT: Bringing machine 'default' up with 'virtualbox' provider...
==> default: Box 'opscode-ubuntu-server-14.04' could not be found. Attempting to find and install...
    default: Box Provider: virtualbox
    default: Box Version: >= 0
==> default: Adding box 'opscode-ubuntu-server-14.04' (v0) for provider: virtualbox
    default: Downloading: https://opscode-vm-bento.s3.amazonaws.com/vagrant/virtualbox/opscode_ubuntu-server-14.04_chef-provisionerless.box

[K    default: Progress: 0% (Rate: 0/s, Estimated time remaining: --:--:--)
[K
STDERR: An error occurred while downloading the remote file. The error
message, if any, is reproduced below. Please fix this error and try
again.

The requested URL returned error: 404 Not Found
---- End output of vagrant up --no-provision --provider=virtualbox ----
Ran vagrant up --no-provision --provider=virtualbox returned 1]
E, [2014-05-21T19:37:40.861827 #4255] ERROR -- Kitchen: ---Nested Exception---
E, [2014-05-21T19:37:40.861883 #4255] ERROR -- Kitchen: Class: Kitchen::ShellOut::ShellCommandFailed
E, [2014-05-21T19:37:40.861939 #4255] ERROR -- Kitchen: Message: Expected process to exit with [0], but received '1'
---- Begin output of vagrant up --no-provision --provider=virtualbox ----
STDOUT: Bringing machine 'default' up with 'virtualbox' provider...
==> default: Box 'opscode-ubuntu-server-14.04' could not be found. Attempting to find and install...
    default: Box Provider: virtualbox
    default: Box Version: >= 0
==> default: Adding box 'opscode-ubuntu-server-14.04' (v0) for provider: virtualbox
    default: Downloading: https://opscode-vm-bento.s3.amazonaws.com/vagrant/virtualbox/opscode_ubuntu-server-14.04_chef-provisionerless.box

[K    default: Progress: 0% (Rate: 0/s, Estimated time remaining: --:--:--)
[K
STDERR: An error occurred while downloading the remote file. The error
message, if any, is reproduced below. Please fix this error and try
again.

The requested URL returned error: 404 Not Found
---- End output of vagrant up --no-provision --provider=virtualbox ----
Ran vagrant up --no-provision --provider=virtualbox returned 1
E, [2014-05-21T19:37:40.861998 #4255] ERROR -- Kitchen: ------Backtrace-------
E, [2014-05-21T19:37:40.862053 #4255] ERROR -- Kitchen: /home/omer/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/test-kitchen-1.2.1/lib/kitchen/shell_out.rb:70:in `rescue in run_command'
E, [2014-05-21T19:37:40.862112 #4255] ERROR -- Kitchen: /home/omer/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/test-kitchen-1.2.1/lib/kitchen/shell_out.rb:59:in `run_command'
E, [2014-05-21T19:37:40.862167 #4255] ERROR -- Kitchen: /home/omer/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/test-kitchen-1.2.1/lib/kitchen/driver/base.rb:158:in `run_command'
E, [2014-05-21T19:37:40.862226 #4255] ERROR -- Kitchen: /home/omer/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/kitchen-vagrant-0.15.0/lib/kitchen/driver/vagrant.rb:123:in `run'
E, [2014-05-21T19:37:40.862282 #4255] ERROR -- Kitchen: /home/omer/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/kitchen-vagrant-0.15.0/lib/kitchen/driver/vagrant.rb:68:in `create'
E, [2014-05-21T19:37:40.862338 #4255] ERROR -- Kitchen: /home/omer/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/test-kitchen-1.2.1/lib/kitchen/instance.rb:273:in `public_send'
E, [2014-05-21T19:37:40.862428 #4255] ERROR -- Kitchen: /home/omer/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/test-kitchen-1.2.1/lib/kitchen/instance.rb:273:in `block in perform_action'
E, [2014-05-21T19:37:40.862484 #4255] ERROR -- Kitchen: /home/omer/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/test-kitchen-1.2.1/lib/kitchen/instance.rb:305:in `call'
E, [2014-05-21T19:37:40.862540 #4255] ERROR -- Kitchen: /home/omer/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/test-kitchen-1.2.1/lib/kitchen/instance.rb:305:in `block in synchronize_or_call'
E, [2014-05-21T19:37:40.862596 #4255] ERROR -- Kitchen: /home/omer/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/test-kitchen-1.2.1/lib/kitchen/instance.rb:303:in `synchronize'
E, [2014-05-21T19:37:40.862653 #4255] ERROR -- Kitchen: /home/omer/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/test-kitchen-1.2.1/lib/kitchen/instance.rb:303:in `synchronize_or_call'
E, [2014-05-21T19:37:40.862708 #4255] ERROR -- Kitchen: /home/omer/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/test-kitchen-1.2.1/lib/kitchen/instance.rb:283:in `block in action'
E, [2014-05-21T19:37:40.862765 #4255] ERROR -- Kitchen: /home/omer/.rbenv/versions/2.1.2/lib/ruby/2.1.0/benchmark.rb:279:in `measure'
E, [2014-05-21T19:37:40.862819 #4255] ERROR -- Kitchen: /home/omer/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/test-kitchen-1.2.1/lib/kitchen/instance.rb:282:in `action'
E, [2014-05-21T19:37:40.862874 #4255] ERROR -- Kitchen: /home/omer/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/test-kitchen-1.2.1/lib/kitchen/instance.rb:273:in `perform_action'
E, [2014-05-21T19:37:40.862929 #4255] ERROR -- Kitchen: /home/omer/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/test-kitchen-1.2.1/lib/kitchen/instance.rb:252:in `create_action'
E, [2014-05-21T19:37:40.862984 #4255] ERROR -- Kitchen: /home/omer/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/test-kitchen-1.2.1/lib/kitchen/instance.rb:246:in `block in transition_to'
E, [2014-05-21T19:37:40.863042 #4255] ERROR -- Kitchen: /home/omer/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/test-kitchen-1.2.1/lib/kitchen/instance.rb:245:in `each'
E, [2014-05-21T19:37:40.863097 #4255] ERROR -- Kitchen: /home/omer/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/test-kitchen-1.2.1/lib/kitchen/instance.rb:245:in `transition_to'
E, [2014-05-21T19:37:40.863152 #4255] ERROR -- Kitchen: /home/omer/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/test-kitchen-1.2.1/lib/kitchen/instance.rb:119:in `converge'
E, [2014-05-21T19:37:40.863206 #4255] ERROR -- Kitchen: /home/omer/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/test-kitchen-1.2.1/lib/kitchen/command.rb:109:in `public_send'
E, [2014-05-21T19:37:40.863261 #4255] ERROR -- Kitchen: /home/omer/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/test-kitchen-1.2.1/lib/kitchen/command.rb:109:in `block (2 levels) in run_action'
E, [2014-05-21T19:37:40.863314 #4255] ERROR -- Kitchen: ----------------------
I, [2014-05-21T19:38:30.149371 #4571]  INFO -- Kitchen: -----> Starting Kitchen (v1.2.1)
I, [2014-05-21T19:38:30.386195 #4571]  INFO -- Kitchen: -----> Creating <default-ubuntu-1404>...
I, [2014-05-21T20:08:50.163444 #4571]  INFO -- Kitchen: -----> Converging <default-ubuntu-1404>...
E, [2014-05-21T20:08:50.182573 #4571] ERROR -- Kitchen: ------Exception-------
E, [2014-05-21T20:08:50.182627 #4571] ERROR -- Kitchen: Class: Kitchen::ActionFailed
E, [2014-05-21T20:08:50.182663 #4571] ERROR -- Kitchen: Message: Failed to complete #converge action: [no implicit conversion of nil into String]
E, [2014-05-21T20:08:50.182960 #4571] ERROR -- Kitchen: ---Nested Exception---
E, [2014-05-21T20:08:50.182992 #4571] ERROR -- Kitchen: Class: TypeError
E, [2014-05-21T20:08:50.183023 #4571] ERROR -- Kitchen: Message: no implicit conversion of nil into String
E, [2014-05-21T20:08:50.183054 #4571] ERROR -- Kitchen: ------Backtrace-------
E, [2014-05-21T20:08:50.183084 #4571] ERROR -- Kitchen: /home/omer/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/kitchen-salt-0.0.16/lib/kitchen/provisioner/salt_solo.rb:333:in `join'
E, [2014-05-21T20:08:50.183157 #4571] ERROR -- Kitchen: /home/omer/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/kitchen-salt-0.0.16/lib/kitchen/provisioner/salt_solo.rb:333:in `prepare_formula'
E, [2014-05-21T20:08:50.183249 #4571] ERROR -- Kitchen: /home/omer/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/kitchen-salt-0.0.16/lib/kitchen/provisioner/salt_solo.rb:151:in `create_sandbox'
E, [2014-05-21T20:08:50.183283 #4571] ERROR -- Kitchen: /home/omer/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/test-kitchen-1.2.1/lib/kitchen/driver/ssh_base.rb:40:in `converge'
E, [2014-05-21T20:08:50.183314 #4571] ERROR -- Kitchen: /home/omer/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/kitchen-vagrant-0.15.0/lib/kitchen/driver/vagrant.rb:75:in `converge'
E, [2014-05-21T20:08:50.183348 #4571] ERROR -- Kitchen: /home/omer/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/test-kitchen-1.2.1/lib/kitchen/instance.rb:273:in `public_send'
E, [2014-05-21T20:08:50.183378 #4571] ERROR -- Kitchen: /home/omer/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/test-kitchen-1.2.1/lib/kitchen/instance.rb:273:in `block in perform_action'
E, [2014-05-21T20:08:50.183410 #4571] ERROR -- Kitchen: /home/omer/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/test-kitchen-1.2.1/lib/kitchen/instance.rb:308:in `call'
E, [2014-05-21T20:08:50.183440 #4571] ERROR -- Kitchen: /home/omer/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/test-kitchen-1.2.1/lib/kitchen/instance.rb:308:in `synchronize_or_call'
E, [2014-05-21T20:08:50.183494 #4571] ERROR -- Kitchen: /home/omer/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/test-kitchen-1.2.1/lib/kitchen/instance.rb:283:in `block in action'
E, [2014-05-21T20:08:50.183739 #4571] ERROR -- Kitchen: /home/omer/.rbenv/versions/2.1.2/lib/ruby/2.1.0/benchmark.rb:279:in `measure'
E, [2014-05-21T20:08:50.183808 #4571] ERROR -- Kitchen: /home/omer/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/test-kitchen-1.2.1/lib/kitchen/instance.rb:282:in `action'
E, [2014-05-21T20:08:50.183884 #4571] ERROR -- Kitchen: /home/omer/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/test-kitchen-1.2.1/lib/kitchen/instance.rb:273:in `perform_action'
E, [2014-05-21T20:08:50.183915 #4571] ERROR -- Kitchen: /home/omer/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/test-kitchen-1.2.1/lib/kitchen/instance.rb:256:in `converge_action'
E, [2014-05-21T20:08:50.183945 #4571] ERROR -- Kitchen: /home/omer/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/test-kitchen-1.2.1/lib/kitchen/instance.rb:246:in `block in transition_to'
E, [2014-05-21T20:08:50.183975 #4571] ERROR -- Kitchen: /home/omer/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/test-kitchen-1.2.1/lib/kitchen/instance.rb:245:in `each'
E, [2014-05-21T20:08:50.184005 #4571] ERROR -- Kitchen: /home/omer/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/test-kitchen-1.2.1/lib/kitchen/instance.rb:245:in `transition_to'
E, [2014-05-21T20:08:50.184034 #4571] ERROR -- Kitchen: /home/omer/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/test-kitchen-1.2.1/lib/kitchen/instance.rb:119:in `converge'
E, [2014-05-21T20:08:50.184073 #4571] ERROR -- Kitchen: /home/omer/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/test-kitchen-1.2.1/lib/kitchen/command.rb:109:in `public_send'
E, [2014-05-21T20:08:50.184103 #4571] ERROR -- Kitchen: /home/omer/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/test-kitchen-1.2.1/lib/kitchen/command.rb:109:in `block (2 levels) in run_action'
E, [2014-05-21T20:08:50.184133 #4571] ERROR -- Kitchen: ----------------------
I, [2014-05-21T20:11:47.409676 #11260]  INFO -- Kitchen: -----> Starting Kitchen (v1.2.1)
I, [2014-05-21T20:11:47.651307 #11260]  INFO -- Kitchen: -----> Converging <default-ubuntu-1404>...
E, [2014-05-21T20:11:47.656079 #11260] ERROR -- Kitchen: ------Exception-------
E, [2014-05-21T20:11:47.656185 #11260] ERROR -- Kitchen: Class: Kitchen::ActionFailed
E, [2014-05-21T20:11:47.656255 #11260] ERROR -- Kitchen: Message: Failed to complete #converge action: [no implicit conversion of nil into String]
E, [2014-05-21T20:11:47.656361 #11260] ERROR -- Kitchen: ---Nested Exception---
E, [2014-05-21T20:11:47.656428 #11260] ERROR -- Kitchen: Class: TypeError
E, [2014-05-21T20:11:47.656491 #11260] ERROR -- Kitchen: Message: no implicit conversion of nil into String
E, [2014-05-21T20:11:47.656553 #11260] ERROR -- Kitchen: ------Backtrace-------
E, [2014-05-21T20:11:47.656614 #11260] ERROR -- Kitchen: /home/omer/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/kitchen-salt-0.0.16/lib/kitchen/provisioner/salt_solo.rb:333:in `join'
E, [2014-05-21T20:11:47.656677 #11260] ERROR -- Kitchen: /home/omer/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/kitchen-salt-0.0.16/lib/kitchen/provisioner/salt_solo.rb:333:in `prepare_formula'
E, [2014-05-21T20:11:47.656740 #11260] ERROR -- Kitchen: /home/omer/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/kitchen-salt-0.0.16/lib/kitchen/provisioner/salt_solo.rb:151:in `create_sandbox'
E, [2014-05-21T20:11:47.656803 #11260] ERROR -- Kitchen: /home/omer/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/test-kitchen-1.2.1/lib/kitchen/driver/ssh_base.rb:40:in `converge'
E, [2014-05-21T20:11:47.656865 #11260] ERROR -- Kitchen: /home/omer/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/kitchen-vagrant-0.15.0/lib/kitchen/driver/vagrant.rb:75:in `converge'
E, [2014-05-21T20:11:47.656927 #11260] ERROR -- Kitchen: /home/omer/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/test-kitchen-1.2.1/lib/kitchen/instance.rb:273:in `public_send'
E, [2014-05-21T20:11:47.656990 #11260] ERROR -- Kitchen: /home/omer/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/test-kitchen-1.2.1/lib/kitchen/instance.rb:273:in `block in perform_action'
E, [2014-05-21T20:11:47.657052 #11260] ERROR -- Kitchen: /home/omer/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/test-kitchen-1.2.1/lib/kitchen/instance.rb:308:in `call'
E, [2014-05-21T20:11:47.657115 #11260] ERROR -- Kitchen: /home/omer/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/test-kitchen-1.2.1/lib/kitchen/instance.rb:308:in `synchronize_or_call'
E, [2014-05-21T20:11:47.657177 #11260] ERROR -- Kitchen: /home/omer/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/test-kitchen-1.2.1/lib/kitchen/instance.rb:283:in `block in action'
E, [2014-05-21T20:11:47.657239 #11260] ERROR -- Kitchen: /home/omer/.rbenv/versions/2.1.2/lib/ruby/2.1.0/benchmark.rb:279:in `measure'
E, [2014-05-21T20:11:47.657301 #11260] ERROR -- Kitchen: /home/omer/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/test-kitchen-1.2.1/lib/kitchen/instance.rb:282:in `action'
E, [2014-05-21T20:11:47.657364 #11260] ERROR -- Kitchen: /home/omer/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/test-kitchen-1.2.1/lib/kitchen/instance.rb:273:in `perform_action'
E, [2014-05-21T20:11:47.657426 #11260] ERROR -- Kitchen: /home/omer/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/test-kitchen-1.2.1/lib/kitchen/instance.rb:256:in `converge_action'
E, [2014-05-21T20:11:47.657489 #11260] ERROR -- Kitchen: /home/omer/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/test-kitchen-1.2.1/lib/kitchen/instance.rb:246:in `block in transition_to'
E, [2014-05-21T20:11:47.657551 #11260] ERROR -- Kitchen: /home/omer/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/test-kitchen-1.2.1/lib/kitchen/instance.rb:245:in `each'
E, [2014-05-21T20:11:47.657613 #11260] ERROR -- Kitchen: /home/omer/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/test-kitchen-1.2.1/lib/kitchen/instance.rb:245:in `transition_to'
E, [2014-05-21T20:11:47.657674 #11260] ERROR -- Kitchen: /home/omer/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/test-kitchen-1.2.1/lib/kitchen/instance.rb:119:in `converge'
E, [2014-05-21T20:11:47.657736 #11260] ERROR -- Kitchen: /home/omer/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/test-kitchen-1.2.1/lib/kitchen/command.rb:109:in `public_send'
E, [2014-05-21T20:11:47.657798 #11260] ERROR -- Kitchen: /home/omer/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/test-kitchen-1.2.1/lib/kitchen/command.rb:109:in `block (2 levels) in run_action'
E, [2014-05-21T20:11:47.657861 #11260] ERROR -- Kitchen: ----------------------
```

**Comment by thedrow** [comment]
Output of kitchen --diagnose:

``` yml
---
timestamp: 2014-05-21 17:14:10.675230807 Z
kitchen_version: 1.2.1
loader:
  process_erb: true
  process_local: true
  process_global: true
  global_config: 
  project_config:
    filename: "/home/omer/Documents/Projects/zBackend/deployment/.kitchen.yml"
    raw_data:
      driver:
        name: vagrant
      provisioner:
        name: salt_solo
      platforms:
      - name: ubuntu-14.04
      suites:
      - name: default
        run_list: 
        attributes: 
  local_config: 
  combined_config:
    filename: 
    raw_data:
      driver:
        name: vagrant
      provisioner:
        name: salt_solo
      platforms:
      - name: ubuntu-14.04
      suites:
      - name: default
        run_list: 
        attributes: 
instances:
  default-ubuntu-1404:
    state_file:
      hostname: 127.0.0.1
      last_action: create
      port: '2222'
      ssh_key: "/home/omer/.vagrant.d/insecure_private_key"
      username: vagrant
    driver:
      box: opscode-ubuntu-14.04
      box_url: https://opscode-vm-bento.s3.amazonaws.com/vagrant/virtualbox/opscode_ubuntu-14.04_chef-provisionerless.box
      customize: {}
      kitchen_root: "/home/omer/Documents/Projects/zBackend/deployment"
      log_level: :info
      name: vagrant
      network: []
      port: 22
      pre_create_command: 
      provider: virtualbox
      sudo: true
      synced_folders: []
      test_base_path: "/home/omer/Documents/Projects/zBackend/deployment/test/integration"
      vagrantfile_erb: "/home/omer/.rbenv/versions/2.1.2/lib/ruby/gems/2.1.0/gems/kitchen-vagrant-0.15.0/templates/Vagrantfile.erb"
      vm_hostname: default-ubuntu-1404.vagrantup.com
    provisioner:
      chef_bootstrap_url: https://www.getchef.com/chef/install.sh
      kitchen_root: "/home/omer/Documents/Projects/zBackend/deployment"
      log_level: :info
      name: salt_solo
      root_path: "/tmp/kitchen"
      salt_apt_repo: http://apt.mccartney.ie
      salt_apt_repo_key: http://apt.mccartney.ie/KEY
      salt_bootstrap_options: ''
      salt_bootstrap_url: http://bootstrap.saltstack.org
      salt_config: "/etc/salt"
      salt_file_root: "/srv/salt"
      salt_install: bootstrap
      salt_minion_config: "/etc/salt/minion"
      salt_pillar_root: "/srv/pillar"
      salt_run_highstate: true
      salt_state_top: "/srv/salt/top.sls"
      salt_version: latest
      state_collection: false
      state_top: {}
      sudo: true
      test_base_path: "/home/omer/Documents/Projects/zBackend/deployment/test/integration"
    busser:
      busser_bin: "/tmp/busser/bin/busser"
      kitchen_root: "/home/omer/Documents/Projects/zBackend/deployment"
      root_path: "/tmp/busser"
      ruby_bindir: "/opt/chef/embedded/bin"
      sudo: true
      suite_name: default
      test_base_path: "/home/omer/Documents/Projects/zBackend/deployment/test/integration"
      version: busser
```

**Comment by thedrow** [comment]
Maybe it's a dup of #2?

**Comment by simonmcc** [comment]
Yeah, related to #2, you're not setting either the formula option or the collection_name option, at least one is needed.  The docs maybe don't explain that very well.

does https://github.com/simonmcc/kitchen-salt/blob/master/provisioner_options.md#collection_name explain it better now?

**Comment by thedrow** [comment]
Yup but can you fix the exception?

**Comment by simonmcc** [comment]
Fixed in v0.0.18, you'll now see an error like this:

```
-----> Converging <default-ubuntu-1204>...
       Preparing files for transfer
       Preparing salt-minion
       Preparing state_top
       Preparing pillars into /srv/pillar
       Preparing state collection
>>>>>> neither collection_name or formula have been set!
>>>>>> ------Exception-------
>>>>>> Class: Kitchen::ActionFailed
>>>>>> Message: Failed to complete #converge action: [exit]
>>>>>> ----------------------
>>>>>> Please see .kitchen/logs/kitchen.log for more details
>>>>>> Also try running `kitchen diagnose --all` for configuration
```
